cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -507,11 +517,49 @@ mkTransaction cfg ((txinFrom, txoutFrom), signingKey) mChangeAddress payments =
   tx = CC.UTxO.UnsafeTx
          { CC.UTxO.txInputs     = txinFrom :| []
          , CC.UTxO.txOutputs    = txOutputs
-         , CC.UTxO.txAttributes = CC.Common.mkAttributes ()
+         , CC.UTxO.txAttributes = createTxAttributes txAdditionalSize
          }
 
   genTx = generalizeTx cfg tx signingKey
 
+-- | If this transaction should contain additional binary blob -
+--   we have to create attributes of the corresponding size.
+--   TxAttributes contains a map from 1-byte integer to arbitrary bytes which
+--   will be used as a binary blob to increase the size of the transaction.
+createTxAttributes
+  :: Maybe TxAdditionalSize
+  -> CC.UTxO.TxAttributes
+createTxAttributes txAdditionalSize =
+  case txAdditionalSize of
+    Nothing -> emptyAttributes
+    Just (TxAdditionalSize size) -> blobAttributes size
+ where
+  emptyAttributes :: CC.UTxO.TxAttributes
+  emptyAttributes = CC.Common.mkAttributes ()
+
+  blobAttributes :: Int -> CC.UTxO.TxAttributes
+  blobAttributes aSize =
+    emptyAttributes {
+      CC.Common.attrRemain = CC.Common.UnparsedFields $
+        Map.singleton k $ LB.replicate finalSize byte
+    }
+   where
```

Comment:
I think removing this nested `where` clause by converting `finalSize` to 
```
finalSize :: Int -> Int64
finalSize aSize = fromIntegral (aSize - sizeOfKey)
```
would make this a lot neater :slightly_smiling_face:

Reply:
Fixed.